server/pulp/server/controllers/repository.py

```diff
@@ -293,15 +294,13 @@ def delete(repo_id):
     # the server logs for more information.
     error_tuples = []  # tuple of failed step and exception arguments
 
-    importer_manager = manager_factory.repo_importer_manager()
     distributor_manager = manager_factory.repo_distributor_manager()
 
     # Inform the importer
-    importer_coll = RepoImporter.get_collection()
-    repo_importer = importer_coll.find_one({'repo_id': repo_id})
+    repo_importer = model.Importer.objects(repo_id=repo_id).first()
```

Comment:
For the other places where we know that the importer does exist or if it doesn't the exception will be caught, `.get()` works well. But in this case, we only need to do something if the importer exists, so it doesn't seem worth catching the `DoesNotExist` and having a try/except inside of another try/except.